dev-docs/npn-and-go-upgrades.md

```diff
@@ -0,0 +1,45 @@
+Taskcluster tries to use the LTS release of Node.js (see
+[release schedule](https://nodejs.org/en/about/releases/)), and the latest
+release of [Go](https://go.dev/). It may be a good idea to update these
+after a Taskcluster release, to get some time with the update in
+development environments before deploying new versions.
+
+# Updating node
+
+Dependabot will open PRs for Node.js, but these will often fail since
+additional steps are needed.
+
+To update the node version:
+
+* Install the new node version, such as `nvm install --lts`
+* Install yarn, such as `npm install -g yarn`
+* Update `package.json`
+* Run `yarn generate`
+* Create a changelog file and commit
+* Build and push new images (see below)
+
+# Updating go
+
+To update the go version:
+
+* Install the new go version, such as `gvm install go1.17.6`
+* Update `.go-version`
+* Run `yarn generate`
+* Create a changelog file and commit
+* Update the package hashes in `workers/generic-worker/gw-decision-task/tasks.yml`
+  with the hashes on the [go downloads page](https://go.dev/dl/)
+* Build and push new images (see below)
```

Comment:
Could you please add to this section the extra steps you took such as https://github.com/taskcluster/taskcluster/pull/5084/commits/970d91d07d2e2490290ae58de00d6e728996cc5e and https://github.com/taskcluster/taskcluster/pull/5084/commits/559a9fea690a5844233592862aa03ab708bdbdc0?

Reply:
Yes. I'm looking for a taskcluster-approved way to run them...

Reply:
I can not find an easy way to run all the formatting stuff, so I added the commands from the build step.